### PR TITLE
feat: [M13] Swift: seg_ids packing-aware paths (chunk mask, boundary conv, block-diagonal attention)

### LINKED ISSUE #263: [M13] Swift: seg_ids packing-aware paths (chunk mask, boundary conv, block-diagonal attention)
Follow-up from #166 (PR #262), which ported the model to mlx-swift for INFERENCE and deliberately left every `seg_ids` arm out.

Missing on the Swift side, all present in `src/model/mlx_backend.py`:

- `_chunk_seg_mask` (`:110-131`) — masks the SSD inter-chunk state carry so recurrent state cannot bleed across a packed document boundary
- `_conv_seq_seg` (`:365-392`) — boundary-aware causal depthwise conv (the conv window is part of the per-doc recurrent state)
- `AttentionBlock.forward_seq`'s block-diagonal mask (`:528-534`)
- `SelectiveSSM.parallel(seg_ids=...)` (`:264-266`)

These matter only when Swift TRAINS on packed corpora, which is #195's territory — inference on a single sequence never needs them. Filed so the omission is recorded rather than rediscovered.

Note the interaction #165 already documents: `parallel(..., return_state=True)` and `seg_ids` are mutually exclusive, because the masked carry-out row reads as zeros.

### PR DESCRIPTION
**Pull request by travisgalloway**
Closes #263

## Summary
- Ports the three `seg_ids` packing-aware forward arms named in #263 from `src/model/mlx_backend.py` to `swift/engine`:
  - `_chunk_seg_mask` -> `SelectiveSSM.chunkSegMask`, applied in `parallel(_:segIds:)`/`scan`
  - `_conv_seq_seg` -> `MambaBlock.convSeqSeg`
  - `AttentionBlock.forward_seq`'s block-diagonal mask
- Widens the `Block` seam (`forwardSeq(_:_:)`) with a `final` one-arg forwarder so every existing call site keeps compiling unchanged; `MoEBlock` accepts and ignores `segIds` (pointwise).
- `parallelWithState`/`forwardPrefill` still take no `seg_ids` (#165's exclusion, unchanged) — now additionally enforced by a `precondition` in `SelectiveSSM.scan` (a masked carry-out would silently read as zeros).
- New `monica-parity` P6 gate against a checked-in `packed.safetensors` oracle (`toy`/`toy-hybrid`/`toy-moe`), exported by a new `--packed-doc-lengths` flag on `scripts/export_parity_fixture.py` using the same chunk-aligned packing rule `src/conformance/doc_boundary_parity.py::check_doc_boundary_parity` already gates in Python. Three assertions per fixture: cross-language logits, per-document self-consistency vs an in-process standalone forward, and an anti-no-op check that the packing-blind forward is provably wrong on the same fixture.
- `tests/test_parity_fixture_export.py` extended to catch `packed.safetensors` staleness the same way it already does for the other checked-in arrays.
- Docs: `swift/engine/Fixtures/README.md` (new "packed (seg_ids) oracle" section) and `docs/design/14-inference-engine.md` (flips the `seg_ids` "not ported" note to RESOLVED, restates the `forwardPrefill` exclusion's #165 rationale).
- Out of scope, matching the plan: training-loop/data-loader `seg_ids` plumbing (#195/#271 territory) and the CUDA backend (already done). `swift/Package.swift` is untouched — no mlx-swift edge added there.

## Testing
- [x] Tests added/updated
- [x] All tests passing
- [x] Manual testing completed

### CODE REVIEW

File: scripts/export_parity_fixture.py [hunk 1]
```diff
@@ -36,11 +36,26 @@
...
+    `"Q,2*Q,5"`) into concrete doc lengths. `Q` is the only name in scope, so this is
+    just enough to express "a chunk multiple, several chunks, and a short/ragged doc" —
+    the exact shapes `tests/test_doc_boundary_parity.py` already gates, so the Swift P6
+    gate and the Python doc-boundary gate stress the same geometry (#68/#263)."""
+    # `eval` here is safe: `spec` is a --packed-doc-lengths CLI argument this maintainer-run
+    # export script's own caller supplies (never untrusted/network input), `__builtins__`
+    # is stripped, and the only name in scope is `Q` — this is arithmetic-expression
+    # parsing, not code execution over attacker-controlled data.
+    ns = {"Q": q}
+    return [int(eval(tok.strip(), {"__builtins__": {}}, ns)) for tok in spec.split(",")]
```
[the remark this answers is not in the document]
Comment (travisgalloway) [reply scripts/export_parity_fixture.py]: ✅ Fixed in d73e644 — replaced `eval()` with an AST-based whitelist parser (`_eval_doc_length_expr`) that only accepts int literals, the name `Q`, and +-*/ operators.

**Comment by travisgalloway** [comment]
Addressed review feedback:

- @copilot-pull-request-reviewer[bot]: overview summary — the one flagged item (unsafe `eval()` in `scripts/export_parity_fixture.py`) is fixed in d73e644 (AST-based whitelist parser); no other actionable content in the summary.
